File: ecubox-backend/src/main/java/com/ecubox/ecubox_backend/dto/LoginResponse.java
package com.ecubox.ecubox_backend.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Schema(description = "Respuesta de autenticación con JWT y permisos del usuario")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LoginResponse {

    @Schema(description = "Token JWT (null en GET /me)", example = "eyJhbGciOiJIUzI1NiIs...")
    private String token;
    @Schema(description = "Nombre de usuario", example = "cliente1")
    private String username;
    @Schema(description = "Correo electrónico", example = "cliente@correo.com")
    private String email;
    @Schema(description = "Fecha de creación de la cuenta")
    private LocalDateTime createdAt;
    @Schema(description = "Roles asignados", example = "[\"CLIENTE\"]")
    private List<String> roles;
    @Schema(description = "Códigos de permisos efectivos")
    private List<String> permissions;
}

File: ecubox_mobile/lib/models/login_response.dart
class LoginResponse {
  const LoginResponse({
    required this.token,
    required this.username,
    this.email,
    this.createdAt,
    this.roles = const [],
    this.permissions = const [],
  });

  final String token;
  final String username;
  final String? email;
  final String? createdAt;
  final List<String> roles;
  final List<String> permissions;

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    final token = json['token'];
    return LoginResponse(
      token: token is String ? token : '',
      username: (json['username'] as String?) ?? '',
      email: json['email'] as String?,
      createdAt: _jsonStringOrNull(json['createdAt']),
      roles: (json['roles'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      permissions:
          (json['permissions'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
    );
  }

  static String? _jsonStringOrNull(Object? value) {
    if (value == null) return null;
    if (value is String) return value;
    return value.toString();
  }
}
